### DISCUSSION TITLE
Server-side apply for CRD in Helm Chart

### DISCUSSION THREAD

**Discussion by ishaan-mehta** [argoproj/argo-cd#20982] [Q&A]
My Helm Chart has a couple CRDs that are too big to sync client-side. To fix, it recommends user server-side syncing or syncing with replace. Using either of these options works via the UI when syncing the problematic CRDs. However, when I try applying the annotation on the CRDs like so:
```
metadata:
  annotations:
    argocd.argoproj.io/sync-options: ServerSideApply=true
```
and I sync (either automatically after deleting the CRD or manually via the UI), it does not server-side apply, and the error comes up again. If I check the SSA box when manually syncing, it works as expected.

What am I missing?

**Comment by aslafy-z** [comment]
Note that `Replace=true` takes precedence over `ServerSideApply=true`. Maybe try `Replace=false,ServerSideApply=true`?

https://argo-cd.readthedocs.io/en/latest/user-guide/sync-options/#server-side-apply
